core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -388,7 +388,7 @@ class Partition(val topic: String,
     val leaderHWIncremented = inWriteLock(leaderIsrUpdateLock) {
       leaderReplicaIfLocal match {
         case Some(leaderReplica) =>
-          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs)
+          val outOfSyncReplicas = getOutOfSyncReplicas(leaderReplica, replicaMaxLagTimeMs).filter(r => !r.ignoreIsrCheck)
```

Comment:
Hi @junrao
in trunk there is no longer a `lastCaughtUpTimeMsUnderlying`

initializing 
```
  @volatile private[this] var _lastCaughtUpTimeMs = time.milliseconds
```
is not enough and our `MetricsDuringTopicCreationDeletionTest` proves that the ISR still may shrink temporarily if the isr-thread runs just after `Replica.updateLogReadResult` has been invoked with `LogReadResult.UnknownLogReadResult`